src/test/java/com/conveyal/datatools/manager/jobs/AutoDeployFeedJobTest.java

```diff
@@ -0,0 +1,155 @@
+package com.conveyal.datatools.manager.jobs;
+
+import com.conveyal.datatools.DatatoolsTest;
+import com.conveyal.datatools.common.status.MonitorableJob;
+import com.conveyal.datatools.manager.auth.Auth0Connection;
+import com.conveyal.datatools.manager.models.*;
+import com.conveyal.datatools.manager.persistence.Persistence;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Date;
+import java.util.List;
+
+import static com.conveyal.datatools.TestUtils.*;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+public class AutoDeployFeedJobTest extends DatatoolsTest {
+    private static final Logger LOG = LoggerFactory.getLogger(AutoDeployFeedJobTest.class);
+    private static OtpServer server;
+    private static Deployment deployment;
+    private static Project project;
+    private static FeedSource mockFeedSource;
+
+    @BeforeClass
+    public static void setUp() throws IOException {
+        DatatoolsTest.setUp();
+        Auth0Connection.setAuthDisabled(true);
+        LOG.info("{} setup", AutoDeployFeedJobTest.class.getSimpleName());
+        String testName = String.format("Test %s", new Date().toString());
+
+        // OTP server instance required by deployment.
+        server = new OtpServer();
+        server.name = testName;
+        Persistence.servers.create(server);
+
+        // deployment.latest() is required to return the latest server id
+        DeployJob.DeploySummary deploySummary = new DeployJob.DeploySummary();
+        deploySummary.serverId  = server.id;
+
+        // a project is required so the pinned deployment id can be defined
+        project = new Project();
+        project.name = testName;
+        Persistence.projects.create(project);
+
+        deployment = new Deployment();
+        deployment.deployJobSummaries.add(deploySummary);
+        deployment.projectId = project.id;
+        Persistence.deployments.create(deployment);
+
+        mockFeedSource = new FeedSource("Mock Feed Source", project.id, FeedRetrievalMethod.MANUALLY_UPLOADED);
+        Persistence.feedSources.create(mockFeedSource);
+    }
+
+    @AfterClass
+    public static void tearDown() {
+        Auth0Connection.setAuthDisabled(Auth0Connection.getDefaultAuthDisabled());
+        server.delete();
+        deployment.delete();
+        project.delete();
+    }
+
+    @Test
+    public void skipJobIfFeedSourceNotDeployable() throws IOException {
```

Comment:
Could we use parametrized tests for `skipJobIfFeedSourceNotDeployable`, `skipJobIfProjectNotAutoDeploy`?

Reply:
@landonreed @binh-dam-ibigroup If I do this update it would probably be best to use junit jupiter the same as otp-middleware, but this will mean adding additional dependencies and migrating from junit to junit jupiter. Thoughts?

Reply:
If that's the case, I don't think including here is warranted. Perhaps we should just log an issue stating our need to upgrade junit and list the particular tests that @binh-dam-ibigroup has mentioned.

Reply:
Issue logged. https://github.com/ibi-group/datatools-server/issues/362